presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoSession.java

```diff
@@ -606,6 +611,32 @@ else if (value instanceof Document) {
         return Optional.ofNullable(typeSignature);
     }
 
+    private String toRemoteSchemaName(String schemaName)
+    {
+        if (!caseInsensitiveNameMatching) {
+            return schemaName;
+        }
+        for (String remoteSchemaName : client.listDatabaseNames()) {
+            if (schemaName.equals(remoteSchemaName.toLowerCase(ENGLISH))) {
+                return remoteSchemaName;
```

Comment:
How expensive is this? how often is it called during planing of a query?

(in JDBC we have caching for this, so question, if we need one here)

Reply:
The first attempt is 3 times and subsequent attempts are only once. 

Finding the last schema takes:
* ~70 msec / 10,000 schemas
* ~700 msec / 10,000 schemas